packages/test/test-service-load/src/faultInjectionDriver.ts

```diff
@@ -158,18 +235,91 @@ extends EventForwarder<IDocumentDeltaConnectionEvents> implements IDocumentDelta
             "error",
             new FaultInjectionError("FaultInjectionError", canRetry));
     }
+
     public injectDisconnect() {
         assert(!this.disposed, "cannot inject disconnect into closed delta connection");
         this.emit("disconnect", "FaultInjectionDisconnect");
     }
+
+    public goOffline() {
+        this.online = false;
+        this.forwarding = false;
```

Comment:
rather than managing two flags, can the base class take a function that just returns the state of online. something like `shouldForward: ()=>boolean`

Reply:
you could also just merge the forwarding logic directly into this class, as its the only one that uses it